sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -124,54 +124,62 @@ public static Map<String, Object> serializeScope(
 
   /**
    * Captures the provided envelope. Compared to {@link IHub#captureEvent(SentryEvent)} this method
-   * - will not enrich events with additional data (e.g. scope) - will not execute beforeSend: it's
-   * up to the caller to take care of this - will not perform any sampling: it's up to the caller to
-   * take care of this - will enrich the envelope with a Session updates is applicable
+   * <br>
+   * - will not enrich events with additional data (e.g. scope)<br>
+   * - will not execute beforeSend: it's up to the caller to take care of this<br>
+   * - will not perform any sampling: it's up to the caller to take care of this<br>
+   * - will enrich the envelope with a Session update if applicable<br>
    *
    * @param envelopeData the serialized envelope data
-   * @return The Id (SentryId object) of the event
-   * @throws Exception In case the provided envelope could not be parsed / is invalid
+   * @return The Id (SentryId object) of the event, or null in case the envelope could not be
+   *     captured
    */
-  public static SentryId captureEnvelope(final @NotNull byte[] envelopeData) throws Exception {
+  @Nullable
+  public static SentryId captureEnvelope(final @NotNull byte[] envelopeData) {
```

Comment:
The return should not be nullable, but rather `SentryId.EMPTY_ID` as every other capture method, so the caller does not have to handle NPEs.